terraform/api_gateway.tf.json.template.py

```diff
@@ -79,14 +80,55 @@ def for_domain(cls, domain):
 
 emit_tf({
     "data": [
-        {
-            "aws_route53_zone": {
-                zone.slug: {
-                    "name": zone.name,
-                    "private_zone": False
+        *(
+            [
+                {
+                    "aws_route53_zone": {
+                        zone.slug: {
+                            "name": zone.name,
+                            "private_zone": False
+                        }
+                    }
+                } for zone in set(zones_by_domain.values())
+            ]
+        ),
+        *(
+            [
+                {
+                    "aws_vpc": {
+                        "gitlab": {
+                            "filter": {
+                                "name": "tag:Name",
+                                "values": ["azul-gitlab"]
+                            }
+                        }
+                    },
+                },
+                {
+                    "aws_subnet": {
+                        f"gitlab_{vpc.subnet_name(public)}_{zone}": {
+                            "filter": {
+                                "name": "tag:Name",
+                                "values": [f"azul-gitlab-{vpc.subnet_name(public)}-{zone}"]
+                            }
+                        }
+                        for public in (False, True)
+                        for zone in range(vpc.num_zones)
+                    }
+                },
+                {
+                    # In order for this to work, the VPC endpoints must have been
+                    # created before the network interface ids can be iterated here.
```

Comment:
This needs to reference the Makefile otherwise it would leave the reader extremely confused.

```suggestion
                    # In order for this to work, the VPC endpoints must have been
                    # created before the network interface IDs can be iterated here.
```

An id is something else.

https://www.merriam-webster.com/dictionary/id